vms/platformvm/validators/manager.go

```diff
@@ -131,7 +131,31 @@ func (m *manager) GetCurrentHeight(context.Context) (uint64, error) {
 	return lastAccepted.Height(), nil
 }
 
-func (m *manager) GetValidatorSet(ctx context.Context, height uint64, subnetID ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+func (m *manager) GetValidatorSet(
+	ctx context.Context,
+	height uint64,
+	subnetID ids.ID,
+) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+	lastAcceptedHeight, err := m.GetCurrentHeight(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if lastAcceptedHeight < height {
+		return nil, database.ErrNotFound
+	}
```

Comment:
Kinda wish we didn't need to call this if the validator set was cached.... But this block should always be cached.